bin/generated_template.d.ts

```diff
@@ -0,0 +1,113 @@
+/** Provides information pointing to a location within a source. */
+export interface Location {
+  /** Line in the parsed source (1-based). */
+  line: number;
+  /** Column in the parsed source (1-based). */
+  column: number;
+  /** Offset in the parsed source (0-based). */
+  offset: number;
+}
+
+/** The `start` and `end` position's of an object within the source. */
+export interface LocationRange {
+  /** Any object that was supplied to the `parse()` call as the `grammarSource` option. */
+  source: any;
+  /** Position at the beginning of the expression. */
+  start: Location;
+  /** Position after the end of the expression. */
+  end: Location;
+}
+
+export interface LiteralExpectation {
+  type: "literal";
+  text: string;
+  ignoreCase: boolean;
+}
+export interface ClassParts extends Array<string | ClassParts> {
```

Comment:
Maybe `ClassRange` instead of `InnerPart`?  We're trying to model a grammar expectation like `[ac-eg]`, which would have ClassParts like `['a', ['c', 'e'], 'g']`.

Reply:
Sure! I gave it a crappy name because I didn't really know what it was for based solely off of getting a superficial understanding of the runtime.

Reply:
I can't seem to mark this as resolved but it can be.